site/blog/2022-10-28-optimize-s3-endpoints.mdx

```diff
@@ -0,0 +1,126 @@
+to control your buckets from internal network with the desired security, and without the extra costs of a
+NAT gateway.
+
+### Why use VPC Interface Endpoints
+
+AWS VPC Interface Endpoints are a component of <a href="https://aws.amazon.com/privatelink">PrivateLink</a> infrastructure. AWS PrivateLink provides private connectivity between virtual private clouds (VPCs),
+supported AWS services, and your on-premises networks withoutexposing your traffic to the public internet.
+
+Relying on PrivateLink offers a set of advantages in terms of **security** and **performance**.
+
+**Security: no internet traversal**
+
+When using endpoints, services will be accessed via Amazon's internal networks to route the traffic. This is
+essential for security services or when hosting any content with sensitive data.
+
+**Security: policies**
+
+Many of the servcies are supporting the use of policies, that can be used to determine the traffic that can
+pass through the interface.
+
+**Performance: latency**
+
+Because of the traffic staying within Amazon’s networks, the physical distance travelled, the number of hops
+and the risk of traversing congested networks are all significantly lower.
+
+**Performance: bandwidth**
+
+PrivateLink supports a sustained bandwidth of 10 Gbps per availability zone with bursts up to 40 Gbps.
+
+**Performance: stability**
+
+PrivateLink consistently lowers the amount of errors and timeouts on high loads
+
+**Do you want to know if you have it properly configured? Please use this query to check it:**
+
+<!--- https://www.urlencoder.org/ -->
+
+<p>
+  <a href="https://app.iasql.com/#/button/Check S3 endpoints/--%20Installing%20the%20needed%20modules%0ASELECT%20iasql_install%28%27aws_s3%27%29%3B%0ASELECT%20iasql_install%28%27aws_vpc%27%29%3B%0A--%20Sync%20the%20existing%20data%0Aselect%20%2A%20from%20iasql_sync%28%29%3B%0A--%20Perform%20the%20query%20for%20endpoints%0Aselect%20bucket.region%2C%20vpc.is_default%2C%20vpc.cidr_block%2C%0A%28select%20count%28%2A%29%3E0%20from%20endpoint_gateway%20where%20endpoint_gateway.region%3Dbucket.region%20and%20service%3D%27s3%27%20and%20endpoint_gateway.vpc_id%3Dvpc.id%29%20as%20has_endpoint_gateway%2C%0A%28select%20count%28%2A%29%3E0%20from%20endpoint_interface%20where%20endpoint_interface.region%3Dbucket.region%20and%20service%3D%27s3%27%20and%20endpoint_interface.vpc_id%3Dvpc.id%29%20as%20has_endpoint_interface%0Afrom%20bucket%20left%20outer%20join%20vpc%20on%20vpc.region%3Dbucket.region%3B">
+    <button class="button button--lg button--primary">Check your endpoints</button>
+  </a>
+</p>
+
+This query will find your active S3 buckets for all regions, associated with the existing endpoint gateways or interfaces - if they exist.
+
+Have you found missing endpoints? No problem, IaSQL can generate them for you. You can create two different
+types of endpoints: gateway and interface. We're going to describe their main features and you could create
```

Comment:
```suggestion
types of endpoints: gateway and interface. We're going to describe their main features and you can create
```

Reply:
I've replaced `could` with `can` in a couple of places because `could` implies an uncertainty that you would do it, while `can` is implying that now you are able to accomplish what you were originally planning to do.

That makes `could` feel like we aren't really endorsing this action and you should take a closer look on if it is the right thing for you to do.